aten/src/ATen/Context.h

```diff
@@ -41,6 +42,7 @@ class TORCH_API Context {
     c10::DeviceType device_type = device.type();
     initCUDAIfNeeded(device_type);
     initHIPIfNeeded(device_type);
+    initHPUIfNeeded(device_type);
```

Comment:
Just a reminder: here will conflict with https://github.com/pytorch/pytorch/pull/136519, please rebase this PR if https://github.com/pytorch/pytorch/pull/136519 is merged first.